packages/middleware-signing/src/middleware.ts

```diff
@@ -10,23 +10,21 @@ import {
 } from "@aws-sdk/types";
 
 import { AwsAuthResolvedConfig } from "./configurations";
-
-const isClockSkewed = (newServerTime: number, systemClockOffset: number) =>
-  Math.abs(getSkewCorrectedDate(systemClockOffset).getTime() - newServerTime) >= 300000;
-
-const getSkewCorrectedDate = (systemClockOffset: number) => new Date(Date.now() + systemClockOffset);
-
-export function awsAuthMiddleware<Input extends object, Output extends object>(
-  options: AwsAuthResolvedConfig
-): FinalizeRequestMiddleware<Input, Output> {
-  return (next: FinalizeHandler<Input, Output>, context: HandlerExecutionContext): FinalizeHandler<Input, Output> =>
+import { getSkewCorrectedDate } from "./utils/getSkewCorrectedDate";
+import { getUpdatedSystemClockOffset } from "./utils/getUpdatedSystemClockOffset";
+
+export const awsAuthMiddleware =
+  <Input extends object, Output extends object>(
+    options: AwsAuthResolvedConfig
+  ): FinalizeRequestMiddleware<Input, Output> =>
+  (next: FinalizeHandler<Input, Output>, context: HandlerExecutionContext): FinalizeHandler<Input, Output> =>
     async function (args: FinalizeHandlerArguments<Input>): Promise<FinalizeHandlerOutput<Output>> {
       if (!HttpRequest.isInstance(args.request)) return next(args);
-      const signer = typeof options.signer === "function" ? await options.signer() : options.signer;
```

Comment:
Why is this removed? Can you confirm why was it added?

Reply:
Looks like it was added in https://github.com/aws/aws-sdk-js-v3/pull/472. Can you make sure we don't rely on it anymore?

Reply:
The check for typeof to be function is removed as it's not required.

The signer is always a Provider in `AwsAuthResolvedConfig`
https://github.com/trivikr/aws-sdk-js-v3/blob/fac5f27fc2727d91be49e403a1a81005efe685d6/packages/middleware-signing/src/configurations.ts#L78-L95

The normalization happens in resolveAwsAuthConfig
 https://github.com/trivikr/aws-sdk-js-v3/blob/fac5f27fc2727d91be49e403a1a81005efe685d6/packages/middleware-signing/src/configurations.ts#L107-L110